rclpy/test/test_executor.py

```diff
@@ -302,6 +303,97 @@ def test_executor_add_node(self):
         assert not executor.add_node(self.node)
         assert id(executor) == id(self.node.executor)
 
+    def test_executor_spin_until_future_complete_timeout(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.003, timer_callback)
+
+        # Timeout
+        future = Future()
+        self.assertFalse(future.done())
+        start = time.monotonic()
+        executor.spin_until_future_complete(future=future, timeout_sec=0.1)
+        end = time.monotonic()
+        self.assertAlmostEqual(end - start, 0.1, places=2)
+        self.assertFalse(future.done())
+
+        timer.cancel()
+
+    def test_excutor_spin_until_future_complete_future_done(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.003, timer_callback)
+
+        def set_future_result(future):
+            time.sleep(0.1)
+            future.set_result('finished')
+
+        # Future complete timeout_sec > 0
+        future = Future()
+        self.assertFalse(future.done())
+        t = threading.Thread(target=lambda: set_future_result(future))
+        t.start()
+        start = time.monotonic()
+        executor.spin_until_future_complete(future=future, timeout_sec=0.2)
+        end = time.monotonic()
+        self.assertAlmostEqual(end - start, 0.1, places=2)
```

Comment:
This test (and the other `assertAlmostEqual` calls) seems flaky, failing in the Windows build:

```
AssertionError: 0.10899999999674037 != 0.1 within 2 places (0.008999999996740365 difference)
```

But it looks like it is within 2 places. Maybe `places=1 is sufficient or instead we could try `delta=0.01`.

Reply:
Fixed